test/batch.js

```diff
@@ -175,11 +206,10 @@ test('batch: deleted way', function (t) {
     var op = {type: 'del', id: 'D', value: { changeset: '4' }}
     db.batch([op], function (err) {
       t.error(err)
-      db.query([[-10, 10], [-10, 10]], function (err, res) {
+      db.query([-10, -10, +10, +10], function (err, res) {
         t.error(err)
-        t.equals(res.length, 1)
-        t.equals(res[0].id, 'D')
-        t.equals(res[0].deleted, true)
+        var ids = res.map(e => e.id).sort()
+        t.deepEquals(ids, ['A', 'B', 'C'])
```

Comment:
if we do it this way (not returning `D`), we'll need to make sure that higher-level modules know that deleting a way => marking the way *and* its nodes as deleted.

cc @gmaclennan for his thoughts

Reply:
I also can't remember if there are problematic implications for not exposing deleted elements in queries (will think on this)